test/test_other.py

```diff
@@ -13308,6 +13308,26 @@ def test_em_js_deps(self):
     ''')
     self.do_runf('f2.c', emcc_args=['f1.c'])
 
+  def test_em_js_deps_cpp(self):
```

Comment:
Do we need a separate test?  Can't we just re-run the existing test_em_js_deps in C++ mode?  (using `@paramaterize`).

Reply:
It does need `namespace {` to actually test mangling. I could add the `namespace {` wrapper under `#ifdef __cplusplus` if that's better but seemed somewhat cleaner to keep it separate. I don't have strong opinion.

Reply:
Does does the test depend on `namespace {`?    Aren't the names mangled either way?

Reply:
Good question. I guess for this kind of variable it's not about mangling but about visibility. Looks like inside `namespace {` the previous implementation was omitted by the linker altogether, so Binaryen didn't find it and did nothing.

Reply:
But also no, wait, it's mangling as well. E.g. try this and disable Output -> Demangle:

https://clang.godbolt.org/z/h9T3bv9Kh

If you comment out `namespace foo {`, the variable is unmangled. I suppose global scope variables are emitted as-is. That's why e.g. top-level `main` also works.

Reply:
Hmm.. I'm a little confused because I think the name used in EM_JS_DEPS in not relevant so mangling should not be relevant.

So the issue is somehow related to the anonymous namespace?

Reply:
Does `extern "C"` somehow also project the variable outside the anonymous namespace and make it a globally visible?

Reply:
Looks like it does:

```
namespace {                                                                      
  int foo;                                                                       
  char* bar;                                                                     
  extern "C" {                                                                   
    int baz;                                                                     
  }                                                                                                               
};   
$ clang++ -c test.cc
$ nm test.o 
0000000000000000 B baz
```

Reply:
> So the issue is somehow related to the anonymous namespace?

Yeah, that's the other thing I mentioned above:

>  I guess for this kind of variable it's not about mangling but about visibility. Looks like inside namespace { the previous implementation was omitted by the linker altogether, so Binaryen didn't find it and did nothing.

Reply:
Yeah fair.

Reply:
Renamed.